ext/hacks.h

```diff
@@ -1,15 +1,26 @@
 #include <vm_core.h>
+#include <version.h>
 
 #define ruby_current_thread ((rb_thread_t *)RTYPEDDATA_DATA(rb_thread_current()))
 
+#if RUBY_API_VERSION_CODE >= 20500
```

Comment:
I'd rather keep the original condition instead of NEW_RUBY, because this name doesn't explain anything and will be obsolete in a year or so